engines/router/missionctl/log/resultlog/bigquery.go

```diff
@@ -42,22 +42,53 @@ func (e *bqLogEntry) Save() (map[string]bigquery.Value, string, error) {
 		return kvPairs, "", errors.Wrapf(err, "Error unmarshaling the result log for save to BQ")
 	}
 
-	// Special handling: Update request.Header to a list of records, expected by BQ.
+	// Special handling: Update request, experiment, enricher, router and ensembler headers to a list of records,
+	// expected by BQ.
 	// It seems protobq.Marshal will be adding support for map[string]string that would help simplify the
 	// implementation of Save().
+	kvPairs["request"] = bigquery.Value(map[string]interface{}{
+		"header": formatBQLogEntryHeader(e.TuringResultLogEntry.Request.Header),
+		"body":   e.TuringResultLogEntry.Request.Body,
+	})
+	kvPairs["experiment"] = formatBQLogEntryResponse(e.TuringResultLogEntry.Experiment)
+	kvPairs["enricher"] = formatBQLogEntryResponse(e.TuringResultLogEntry.Enricher)
+	kvPairs["router"] = formatBQLogEntryResponse(e.TuringResultLogEntry.Router)
+	kvPairs["ensembler"] = formatBQLogEntryResponse(e.TuringResultLogEntry.Ensembler)
```

Comment:
I've added these additional lines after realising that the headers in the response need to be specially handled before being pushed to BigQuery tables. There's a similar step done for the request headers above.